### ISSUE TITLE
Sea Block Mod Set needs updating

### ISSUE THREAD

**Issue by KiwiHawk** [factoriolab/factorio-lab#700]
https://mods.factorio.com/mod/SeaBlockMetaPack

The mod list for the Sea Block preset should be Sea Block Pack mod and it's dependencies. With default settings. The current mod set that you have here has a LOT of recipes disabled that should be available. It also seems to include mods that aren't part of the sea block pack (Bob's Greenhouses, Angel's Industries, Bob's Character Classes).

Most but not all of the recipes listed should be enabled. There are a few that are correctly disabled (e.g. Iron rod stack, Steel rod stack, Manganese sheet coil).

I tried manually enabling the recipes that should be available. I told it to produce red science. It found a solution eventually but took a very long time to do so. Is this normal?

Several of the ores are displaying the wrong name. It shows "Galena (Lead Ore)", "Rutile (Titanium Ore)", "Quartz (Silicon Ore)", "Bauxite (Aluminium Ore)". In Factorio, only the name in brackets is shown.

I'm the current dev for Sea Block; took over maintenance and updates from Trainwreck the year before last. Let me know if there's any other info I can give you.

I've had people asking if there was a mobile accessible calculator that could handle Sea Block. Thought I should give this a try before recommending it. Hope these issues get resolved so I am able to recommend it! :-)

**Comment by dcbroad3** [comment]
Thanks for reaching out!
I can try to resolve most of these issues. I think currently this set includes sea block plus all the rest of bobs and angels, it would be easy enough to reduce it to only the required dependencies. 

The calculator is still sometimes slow. Often the number of recipes available is much larger than you'd think, and it takes a long time to solve. That's part of why it defaults to having many recipes disabled, it is faster if it can ignore things. I'm sure there is room for improvement in the defaults.

**Comment by dcbroad3** [comment]
@KiwiHawk I'm creating a PR with an updated data set for Sea Block that will be pushed to https://factoriolab.github.io/staging momentarily. Please check it out and let me know what you think. A few things have not been addressed. 

* The same set of recipes are disabled by default. I can update this list, but as you note it does sometimes cause the calculator to be slow. I would advise against enabling most or all recipes. Enabling all recipes and attempting to solve for red science, I can see that the matrix is 1003 x 1650, and the solver completed 148 pivots in the first 5 seconds. I plan to implement a WebAssembly matrix solver in #719 that should improve performance, but I can't say by how much until I've tried it out. 
* You mentioned several ores display the wrong name. I am using a utility to build data sets, source code here: https://github.com/factoriolab/factorio-lab-tools. I did not originally author the tool and there are definitely bugs including bugs in getting the correct names of items and recipes. I hope to move away from using that tool at some point and use a built in mod to build my data sets but then I have a problem building the icons / sprite sheets as that's not possible via a mod. I would rather spend effort building a new data generator at this point in time than to fix the existing tool. 

Let me know your thoughts, I am more than happy to make adjustments to the data as you see fit!

**Comment by dcbroad3** [comment]
Created a brand new issue for the new data generator, #737, as it's been sitting in the back of my mind for long enough now.

**Comment by KiwiHawk** [comment]
Had a quick look. Seem right from what I can tell by looking at the data.

I did notice an issue with the icons. Once again, a library issue probably. The overlay over the barrel / gas bottle icons should be coloured. Different tints for each part.
![image](https://user-images.githubusercontent.com/59639/168023871-1f1d56f2-ad92-4226-86e2-e95e1e34d2b4.png)


Expected:
![image](https://user-images.githubusercontent.com/59639/168024711-6b6acaba-dbfb-4fd0-bd08-cf95ee4135ad.png)

**Comment by dcbroad3** [comment]
Good call. I hadn't noticed before that the tint object could be an array of rgb values. I think that should be fixed now, new icon sheet is pushing to the same staging site if you can verify.

**Comment by KiwiHawk** [comment]
## Disabled Items are Showing

![image](https://user-images.githubusercontent.com/59639/168403825-7b676ac9-04f2-479d-a234-41e661b5e6f5.png)
![image](https://user-images.githubusercontent.com/59639/168404320-78153643-9249-45be-a109-f0e590fa825c.png)
![image](https://user-images.githubusercontent.com/59639/168404062-84a2e5dc-e3a1-4bac-a14b-e06c199f37fd.png)

Pollution cleaning and pollution producing boards shouldn't show up. These were disabled by the latest version of Circuit Processing mod (0.4.4)

- Pollution cleaning circuit board
- Pollution cleaning logic board
- Pollution cleaning processor board
- Pollution producing circuit board
- Pollution producing logic board
- Pollution producing processor board
- Silicon boule
- Cobalt sheet coil
- Cobalt plate
- Iron rod stack
- Manganese sheet coil
- Manganese plate
- Nickel sheet roll
- Nickel plate
- Zinc sheet coil
- Zinc plate
- Steel rod stack
- Hogger refugium

## Incorrect Item / Fluid Names
![image](https://user-images.githubusercontent.com/59639/168404236-232c0bd3-369d-40a5-b0b2-6de64301ca70.png)
![image](https://user-images.githubusercontent.com/59639/168404695-352d0452-ab60-48ad-a286-bf869ae3696f.png)
![image](https://user-images.githubusercontent.com/59639/168404709-64d55fa2-a7ef-4584-9626-c861614a0e06.png)

- 7.5% Semiheavy water
- 12.5% Semiheavy water
- 15% Semiheavy water
- Spidertron Cannon


## Other
![image](https://user-images.githubusercontent.com/59639/168404875-a66081fc-6d76-4461-a055-9145a8a5c4a4.png)

Unavailable recipes are listed under disabled recipes. They aren't listed anywhere else in the file. I've circled a couple of examples.
No idea if this is an issue or not. Fixing it might speed things up slightly?

**Comment by dcbroad3** [comment]
Thanks for taking the time to write this up! I'll see what I can do to resolve these. 
I've also started attempting to write an entirely new data parser, but that has a long way to go and still not sure yet if it'll be feasible to completely replace the old parser.

**Comment by dcbroad3** [comment]
> Disabled Items are Showing
This should generally be fixed now. The one exception is is 'Silicon boule'. For reasons I cannot determine the lua loader I am using is not swapping the basic `silicon` for the angels silicon. I can see the code that is supposed to swap it in the Lua, I can see it is swapped in game, but I don't know why it isn't swapping in my data loader. I may investigate further later but I'd probably rather try to fix that in data loader 2.0. 

> Incorrect Item / Fluid Names
I plan to fix these things in data loader 2.0. The current data parser is reading and interpolating into localized strings directly, and clearly has bugs. If I can get a data loader as a mod it can translate the same way the game does and be much more reliable. If, for some reason, that does not work out, I will revisit this at that point and see if I can fix the parsing of names.

> Other
I made sure to filter the disabled recipes for only recipes that are in the data set, and added some development-mode code to check that list every time so I'll be warned in the future.

**Comment by dcbroad3** [comment]
I'm leaving myself a note to check with you when the new data generator is usable to verify that the remaining issues are resolved. I think I've worked out all the problems with whether a mod data generator is feasible now, just need to finish building the thing. I'm going to close this issue as part of the already open PR.

I've already got the new generator running inside the game to get names and recipes, so it should have much less trouble getting those remaining issues resolved.
